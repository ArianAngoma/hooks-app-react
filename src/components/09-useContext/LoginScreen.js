import {useContext} from "react";
import {UserContext} from "./UserContext";

const LoginScreen = () => {
    const {setUser} = useContext(UserContext);

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>

            <button className='btn btn-primary'
                    onClick={() => {
                        setUser({
                            id: 1,
                            name: 'Arian'
                        })
                    }}>
                login
            </button>
        </div>
    )
}

export default LoginScreen;