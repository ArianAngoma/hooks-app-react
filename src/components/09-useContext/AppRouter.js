import {
    BrowserRouter as Router,
    Switch,
    Route, Redirect,
    // Link
} from "react-router-dom";
import NavBar from "./NavBar";
import AboutScreen from "./AboutScreen";
import LoginScreen from "./LoginScreen";
import HomeScreen from "./HomeScreen";

const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar/>
                <Switch>
                    <Route exact path='/' component={HomeScreen}/>
                    <Route exact path='/about' component={AboutScreen}/>
                    <Route exact path='/login' component={LoginScreen}/>

                    {/*<Route component={HomeScreen}/>*/}
                    <Redirect to="/"/>
                </Switch>
            </div>
        </Router>
    )
}

export default AppRouter;