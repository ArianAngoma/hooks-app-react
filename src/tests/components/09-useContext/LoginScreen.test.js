import {mount, shallow} from "enzyme";
import LoginScreen from "../../../components/09-useContext/LoginScreen";
import {UserContext} from "../../../components/09-useContext/UserContext";

describe('Test on <LoginScreen/>', () => {
    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={{setUser}}>
            <LoginScreen/>
        </UserContext.Provider>
    );

    test('Should display correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should run setUser with the expected argument', () => {
        wrapper.find('button').prop('onClick')();

        expect(setUser).toHaveBeenCalledWith({
            id: 1,
            name: 'Arian'
        });
    });
})