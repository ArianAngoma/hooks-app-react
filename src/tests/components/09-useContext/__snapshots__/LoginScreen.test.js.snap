// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <LoginScreen/> Should display correctly 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    login
  </button>
</div>
`;
